modin/pandas/base.py

```diff
@@ -122,6 +122,24 @@ class BasePandasDataset(ClassLogger):
     # but lives in "pandas" namespace.
     _pandas_class = pandas.core.generic.NDFrame
 
+    # TODO(https://github.com/modin-project/modin/issues/4821):
+    # make this cache_readonly
+    @property
```

Comment:
`cache_readonly` is accessible from `pandas.util`, why cannot we just import it and use instead of adding a TODO? It's not specific to any Python version (as I'm _not_ suggesting to use `cached_property` from `functools`, as I find pandas version safer even if a little slower)